### PR TITLE
fixes to Javadoc

### PR DESCRIPTION
**Pull request by gavinking**
- fallout from removal of type variable from `PageRequest`
- removal of some uses of `{@code }` inside `<pre>`, which renders  extremely weirdly in IntelliJ
- improve description of ordering in `@OrderBy`

### CODE REVIEW

**Review by otaviojava — APPROVED** [review]

File: api/src/main/java/jakarta/data/Order.java [hunk 1]
```diff
@@ -37,14 +37,13 @@
  * sort criteria during development. For example,</p>
  *
  * <pre>
- * {@code Page<Employee>} findByYearHired(int year, {@code PageRequest<Employee>} pageRequest);
+ * Page&lt;Employee>&gt; findByYearHired(int year, PageRequest pageRequest, Order&lt;Employee&gt;);
  * ...
  * page1 = employees.findByYearHired(Year.now(),
+ *                                   PageRequest.ofSize(10).page(1)
```
Comment (njr-11) [inline api/src/main/java/jakarta/data/Order.java:43]:
```suggestion
 *                                   PageRequest.ofSize(10).page(1),
```

File: api/src/main/java/jakarta/data/Order.java [hunk 2]
```diff
@@ -37,14 +37,13 @@
...
  *                                   Order.by(_Employee.salary.desc(),
  *                                            _Employee.lastName.asc(),
- *                                            _Employee.firstName.asc())
- *                                        .page(1)
- *                                        .size(10));
+ *                                            _Employee.firstName.asc());
```
Comment (njr-11) [inline api/src/main/java/jakarta/data/Order.java:46]:
```suggestion
 *                                            _Employee.firstName.asc()));
```

File: api/src/main/java/jakarta/data/page/Page.java [hunk 3]
```diff
@@ -28,11 +28,11 @@
...
- * {@code @Find}
- * {@code Page<Vehicle>} search({@code @By("make")} String make,
- *                       {@code @By("model")} String model,
- *                       {@code @By("year")} int designYear,
- *                       {@code PageRequest<Vehicle>} pageRequest);
+ * &#64;Find
+ * Page&lt;Vehicle&gt; search(&#64;By("make") String make,
+ *                      &#64;By("model") String model,
+ *                      &#64;By("year") int designYear,
+ *                      PageRequest pageRequest);
```
Comment (njr-11) [inline api/src/main/java/jakarta/data/page/Page.java:35]: We should never have examples of pagination that lack a way to provide a deterministic ordering which is needed for it to work properly
```suggestion
 * Page&lt;Vehicle&gt; search(&#64;By("make") String make,
 *                      &#64;By("model") String model,
 *                      &#64;By("year") int designYear,
 *                      PageRequest pageRequest,
 *                      Order&lt;Vehicle&gt; order);
```

File: api/src/main/java/jakarta/data/repository/OrderBy.java [hunk 4]
```diff
@@ -44,13 +44,19 @@
  * <pre>
  * &#64;OrderBy("lastName")
  * &#64;OrderBy("firstName")
- * Person[] findByZipCode(int zipCode, {@code PageRequest<?>} pageRequest);
+ * Person[] findByZipCode(int zipCode, PageRequest pageRequest);
```
Comment (njr-11) [inline api/src/main/java/jakarta/data/repository/OrderBy.java:47]:
```suggestion
 * &#64;OrderBy("id")
 * Person[] findByZipCode(int zipCode, PageRequest pageRequest);
```

File: api/src/main/java/jakarta/data/repository/Param.java [hunk 5]
```diff
@@ -29,12 +29,12 @@
...
  * <pre>
- * {@code @Repository}
- * public interface Products extends BasicRepository{@code <Product, String>} {
+ * &#64;Repository
+ * public interface Products extends BasicRepository&lt;Product, String&gt; {
  *
- *     {@code @Query("WHERE length * width * height <= :maxVolume")}
- *     {@code Page<Product>} freeShippingEligible({@code @Param}("maxVolume") float volumeLimit,
- *                                        {@code PageRequest<?>} pageRequest);
+ *     &#64;Query("WHERE length * width * height <= :maxVolume ORDER BY price ASC")
```
Comment (njr-11) [inline api/src/main/java/jakarta/data/repository/Param.java:35]:
```suggestion
 *     &#64;Query("WHERE length * width * height <= :maxVolume ORDER BY price ASC, id ASC")
```

File: api/src/main/java/jakarta/data/repository/Query.java [hunk 6]
```diff
@@ -74,26 +74,27 @@
...
  *     // JPQL using named parameters
- *     {@code @Query("select distinct name from Person where length(name) >= :min and length(name) <= :max")}
- *     {@code Page<String>} namesOfLength({@code @Param}("min") int minLength,
- *                                {@code @Param}("max") int maxLength,
- *                                {@code PageRequest<Person>} pageRequest);
+ *     &#64;Query("select distinct name from Person " +
+ *            "where length(name) >= :min and length(name) <= :max")
+ *     Page&lt;String&gt; namesOfLength(&#64;Param("min") int minLength,
+ *                                &#64;Param("max") int maxLength,
+ *                                pageRequest);
```
Comment (njr-11) [inline api/src/main/java/jakarta/data/repository/Query.java:97]:
```suggestion
 *                                PageRequest pageRequest,
 *                                Order&lt;Person&gt; order);
```

File: api/src/main/java/module-info.java [hunk 7]
```diff
@@ -773,49 +773,41 @@
...
  * <h3>Sorting</h3>
  *
  * <p>When a page is requested with a {@code PageRequest}, dynamic sorting
  * criteria may be supplied by passing instances of {@link Sort} or {@link Order}.
- * and its overloads. For example,</p>
+ * For example,</p>
  *
  * <pre>
- * Product[] findByNameLike(String pattern, {@code PageRequest<Product>} pagination);
+ * Product[] findByNameLike(String pattern, PageRequest page, Order&lt;Product&gt; order);
```
Comment (njr-11) [inline api/src/main/java/module-info.java:789]:
```suggestion
 * Product[] findByNameLike(String pattern, PageRequest pageRequest, Order&lt;Product&gt; order);
```

**Comment by gavinking** [comment]
Applied all suggestions.

**Review by njr-11 — APPROVED** [review]

**Pull request merged**